### PR TITLE
[NayNay] Help text for command options

### LINKED ISSUE #244: 🔥  `entropy -a --help` does not help and nukes config
if you run the command above it will nuke your config. proceed with caution... and backups

### PR DESCRIPTION
**Pull request by rh0delta**
## Related Issue(s)

- Resolves #244 

## Proposed Changes

- contrary to intended use of the help option with commander, this PR adds handler when user adds help option after option that requires an argument
- for now, only updated account option, may require discussion with team if this is how we want to proceed

## Testing

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Manual testing performed

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have added tests.
- [ ] I have commented my code.
- [ ] I have included a `CHANGELOG.md` entry.
- [ ] I have updated documentation in `github.com:entropyxyz/entropy-docs`, where necessary.

### CODE REVIEW

**Comment by mixmix** [comment]
@rh0delta says this needs more design work
Doesn't need to be in this release

Call on someone to discuss options e.g. Mix

**Comment by mixmix** [comment]
discussed : should throw on `-a --help`

File: src/common/utils-cli.ts [hunk 1]
```diff
@@ -58,6 +58,16 @@ export function accountOption () {
       // We try to map addressOrName to an account we have stored
       if (!storedConfig) return addressOrName
 
+      const availableAccounts = {}
+      storedConfig.accounts.forEach(acct => availableAccounts[acct.name] = acct.address)
+      if (!Object.keys(availableAccounts).includes(addressOrName) || !Object.values(availableAccounts).includes(addressOrName)) {
```
Comment (mixmix) [inline src/common/utils-cli.ts:63]: :fire:  this is `addressOrName` so this check is not a good match.

`findAccountByAddressOrName` already handles the lookup gracefully. I think you wanna modify `if (!account) { }`

File: src/common/utils-cli.ts [hunk 2]
```diff
@@ -58,6 +58,16 @@ export function accountOption () {
...
+        print('=============AVAILABLE ACCOUNTS==============')
+        print(availableAccounts)
+        print('=============================================\n')
+        console.error(`AccountError: [${addressOrName}] is not a valid argument for the account option. Please use one of the aliases or addresses as shown above\n`)
+        process.exit(1)
+      }
+
       const account = findAccountByAddressOrName(storedConfig.accounts, addressOrName)
       if (!account) return addressOrName
```
Comment (mixmix) [inline src/common/utils-cli.ts:72]:
```suggestion
      if (!account) {
        console.error(`AccountError: [${addressOrName}] is not a valid argument for the account option.`)
        // if want to print accounts use some existing tool that formats them in the code?
        process.exit(1)
      }
```

Comment (rh0delta) [reply src/common/utils-cli.ts]: updated to use this format, but kept the log of the account and addresses there

**Review by mixmix — APPROVED** [review]

**Comment by mixmix** [comment]
Looks good to me - it meets the needs of the issue. We could print out more but your help text is plenty helpful IMO

File: src/common/utils-cli.ts [hunk 3]
```diff
@@ -59,7 +59,11 @@ export function accountOption () {
       if (!storedConfig) return addressOrName
 
       const account = findAccountByAddressOrName(storedConfig.accounts, addressOrName)
-      if (!account) return addressOrName
+      if (!account) {
+        console.error(`AccountError: [${addressOrName}] is not a valid argument for the account option.`)
+        print(bold('!! Available accounts can be found using entropy account list || entropy account ls !!'))
+        process.exit(1)
```
Comment (mixmix) [inline src/common/utils-cli.ts:65]: hmmm..  wonder why CI is breaking ...surely this doesn't need a `return` as well as `process.exit`?
The error in CI was "broken config"

Comment (mixmix) [reply src/common/utils-cli.ts:65]: Could very well be a different issue / race

Comment (rh0delta) [reply src/common/utils-cli.ts:65]: it's weird, it happens intermittently, but it happens on the program deploy method. might be a race condition with the tmp config file. don't think it needs to be completely solved now as it doesn't break cli

Comment (rh0delta) [reply src/common/utils-cli.ts:65]: re-running the job results in a pass

**Pull request merged**